src/v2entities/trade.ts

```diff
@@ -334,14 +336,19 @@ export class TradeV2 {
     routes: RouteV2[],
     tokenAmountIn: TokenAmount,
     tokenOut: Token,
+    isAvaxIn: boolean,
+    isAvaxOut: boolean,
     provider: Provider | Web3Provider | any,
     chainId: ChainId
   ): Promise<Array<TradeV2 | undefined>> {
     const isExactIn = true
-    const isAvaxIn = tokenAmountIn.token.address === WAVAX[chainId].address
-    const isAvaxOut = tokenOut.address === WAVAX[chainId].address
 
-    if (isAvaxIn && isAvaxOut) {
+    // handle wavax<->avax wrap swaps
+    const isWrapSwap =
+      (isAvaxIn && tokenOut.address === WAVAX[chainId].address) ||
+      (isAvaxOut && tokenAmountIn.token.address === WAVAX[chainId].address)
+
+    if (isWrapSwap) {
```

Comment:
Previously, I made the wrong assumption that "WAVAX" tokens could just be treated as AVAX. 

So instead we pass-in `isAvaxIn` and `isAvaxOut` as a parameter to this function. The frontend is expected to make the determination whether we are swapping AVAX or WAVAX tokens.